### DISCUSSION TITLE
Is Rustdesk Same as rudesktop?

### DISCUSSION THREAD

**Discussion by penangit** [rustdesk/rustdesk#4981] [General]
I stumbled upon rudesktop.ru everything looks the same in the images as to rustdesk

https://rudesktop.ru/

**Comment by rustdesk** [comment]
They are the clone.

**Reply by dinger1986** [reply]
As in rudesktop has cloned rustdesk and did you see the prices?

**Reply by SergeyMy** [reply]
the question is not about prices, but about the fact that they violate the GPL! who would tell them https://sfconservancy.org/copyleft-compliance

**Reply by dinger1986** [reply]
Oh the license violation is the major concern. Wouldn't believe how much it happens! Thanks for flagging it up

**Reply by SergeyMy** [reply]
нужен кто-то из России с **активной электронной подписью** - и эту компанию вышибут отовсюду.
как минимум из "Реестр российского программного обеспечения" уж точно.
достаточно подать заявление о нарушении.

**Reply by penangit** [reply]
Interesting

**Reply by alex-eri** [reply]
@SergeyMy так как GPL опирается на американские законы - она тут не особо годится чтоб просто выпереть. Нужна заявка от одного из авторов rustdesk из России, который при этом не работает паралельно в rudesktop.

**Reply by berkut116** [reply]
> нужен кто-то из России с **активной электронной подписью** - и эту компанию вышибут отовсюду. как минимум из "Реестр российского программного обеспечения" уж точно. достаточно подачи заявления о нарушении.

Это так не работает .  у нас чудная страна.даже debian продают  с новой оболочкой

**Comment by polyzium** [comment]
Hi, I just discovered this. I'm from Russia and it disgusts me to see software that is blatantly stolen.

Note how the screenshots look eerily similar (if not identical) to RustDesk, almost one-to-one. I also could not find any mention of RustDesk anywhere on their website.
They claim that this is "indigenous software", which sounds like propaganda. They also seem to have entries in the Unified Registry of Russian Software and some sort of "compatibility licenses". All this to show how "proud" the developers are.

I think the RustDesk team should launch an investigation into the company responsible and file a lawsuit/DMCA takedown notice.

**Reply by DJRaov** [reply]
> I think the RustDesk team should launch an investigation into the company responsible and file a lawsuit/DMCA takedown notice.

Not that the Russian government would do much, to be honest. I haven't heard of them complying with DMCAs at least since the 2022 conflict, let alone foreign lawsuits since the government does not even acknowledge their existance. The action should be taken from _within_ the country for it to have any effect I think.

**Reply by SergeyMy** [reply]
We need someone from Russia with an active electronic signature

and the appeal will be tantamount to going to court!

**Reply by polyzium** [reply]
> I haven't heard of them complying with DMCAs at least since the 2022 conflict

I will be pissed if they go the way of China, where copyright laws are basically non-existent.

> We need someone from Russia with an active electronic signature

I am not sure that one person sacrificing himself for this is a good idea, maybe a class action lawsuit would be a better option?

**Reply by SergeyMy** [reply]
In Russia, copyrights are respected, often more strictly than elsewhere.

PS the discussion turns into a political aspect, this is usually unacceptable.

**Comment by ITR-Just4Work** [comment]
Простите уж за мой чёрствый язык, но какие же они грёбанные уёбыши.
Сервера RustDesk замедляют/блочат, а свои продвигают.

**Reply by pidrilaebanaja** [reply]
благо можно поднять свой сервер rustdesk'а

**Reply by ermak-dev** [reply]
Бро, ты так говоришь будто мы РКН. Мы обычные пацаны из Новосибирска, которые сделали продукт сначала для себя, что бы тимвьювер заменить, никакого отношения к государству не имеем.

**Reply by ethaniel** [reply]
> которые сделали продукт 

Бро, ты глаголом ошибся.

А вообще тут скорее "Пацаны шли к успеху", где продолжение вы узнаете в ближайшие недели.

**Reply by ermak-dev** [reply]
Don't Be a Menace to South Central While Drinking Your Juice in the Hood

**Comment by ethaniel** [comment]
## $2,600,000 is the gross income of that company in 2025. There have been 0 lawsuits against that company. 
ООО "Передовые Технологии", VAT # 9703094665: https://www.rusprofile.ru/id/1227700375891

Company shareholders (%):
**Ermak Anton Anatolievich** - Ермак Антон Анатольевич (42,5%) -- @ermak-dev
**Kandibovich Dmitriy Petrovich** - Кандыбович Дмитрий Петрович (42,5%)
**Semenova Natela Otarovna** - Семенова Натела Отаровна (10%)
**Molin Ruslan Alekseevich** - Молин Руслан Алексеевич (5%)

<img width="579" height="528" alt="image" src="https://github.com/user-attachments/assets/771a704d-6114-47d9-90b9-23ba8d2a3e19" />

**Comment by ermak-dev** [comment]
Thanks to everyone who raised this. There are some strong claims in this thread that RuDesktop violates the AGPLv3, and we'd rather answer them in the open than let them sit. Here's the whole picture.

First, why RuDesktop isn't under the AGPL. It comes down to Sciter, and this is the part that keeps getting missed. RustDesk depends on the Sciter engine (sciter.dll / .dylib / .so), which ships under a proprietary EULA from Terra Informatica Software, Inc. That EULA is incompatible with the AGPLv3. The AGPLv3 requires every component of a distributed product to be licensed under terms that allow free redistribution and source access, and a closed-source engine doesn't clear that bar. So RustDesk itself can't be distributed as a fully AGPLv3-compliant product. And a project that can't satisfy the AGPLv3 upstream can't impose AGPLv3 obligations downstream that it doesn't meet itself.

In our case, we hold a commercial Sciter license, linked into RuDesktop. That's fully compliant with our agreement with Terra Informatica. But you can't relicense a proprietary library under the AGPLv3. So releasing RuDesktop under the AGPL was never legally possible in the first place. That's the real reason, not an attempt to dodge anything.

On the server and the "fork" accusation. The RuDesktop server was written from scratch. It uses its own network protocols and its own encryption, both distinct from RustDesk's. We built it when RustDesk's server components were still closed source, before 1.1.6. The AGPLv3 obligations being cited here (source disclosure, license consistency, copyright notices) apply to derivative works of AGPLv3-licensed code. Our server predates the public release of RustDesk's server source and is an independent implementation, so it isn't a derivative work. If anyone has specific technical evidence to the contrary, we'll look at it. But "it does the same thing" isn't evidence of copied code.

As for where this is going: we're rewriting the GUI on Slint to drop Sciter entirely. Once that's done, the proprietary-license barrier above goes away, and we plan to publish a stripped-down open-source edition of RuDesktop under the AGPL. Nothing requires us to. We'd like to give something back where we reasonably can, and it's the cleanest way to settle this.

Happy to keep going on specifics with anyone arguing in good faith. If you have concrete evidence rather than assumptions, bring it and we'll engage with it directly.

**Reply by ethaniel** [reply]
> We'd like to give something back where we reasonably can, and it's the cleanest way to settle this.

Wait, so let me get this straight. You guys copy/pasted Rustdesk's source code and completely violated the AGPLv3 by claiming that it is a "fully russian product", made millions of dollars and now saying that you'd like to give something back? That's not how the license world works.

**Comment by ethaniel** [comment]
If anyone else is involved in this, and have already collected the evidence, please do not disclose it here.

**Comment by ermak-dev** [comment]
To put an end to the back and forth, we've done the most direct thing we can: the parts of the client that derive from RustDesk are now published under the AGPLv3, as of today.

**https://sourcecraft.dev/rudesktop/rudesktop-client**

The corresponding source, the RustDesk copyright notices and the license text are all in place, and the tree builds. If you spot derived code that's missing from the disclosure, point to it and we'll add it.

On the "copy/pasted, it's just a rebrand" claim, that's a separate question from licensing, and the feature list answers it on its own. Here's everything in RuDesktop that goes beyond RustDesk's free core, in two groups.

---

## A) RustDesk has an equivalent, but only in the paid Server Pro edition

**Remote session**
- Session recording (video and file transfer)
- Wake-on-LAN

**Administration and infrastructure**
- Web admin console (Django + React)
- State backend on a real database (PostgreSQL + Redis)
- REST API for devices, groups, jobs, policies and queries

**Identity and access**
- Active Directory / LDAP / OIDC
- MFA / two-factor
- Role-based ACLs and granular per-connection-type permissions

**Compliance**
- Centralized audit logging (CEF format for SIEM)

---

## B) Not in RustDesk at all, neither OSS nor Server Pro

**Remote session**
- USB-over-IP (forward remote USB devices)
- Remote printing (printer redirection)
- In-session voice calls
- Shareable invite links with expiry and type/device/user restrictions

**MDM for Android**
- QR-code enrollment
- Wi-Fi provisioning during enrollment (SSID/password/WPA/WEP/EAP)
- Device Admin component plus MDM JSON config bundle
- Remote lock and remote wipe
- Kiosk mode
- App install/uninstall control
- Restrictions for factory reset, debug, developer mode and system apps
- Hardware controls for camera, Bluetooth, roaming, brightness, volume, screen timeout, USB storage and stay-on-while-plugged
- Device geo-tracking and geo-fencing

**UEM**
- Job and automation engine (priorities, timeouts, run-on-start, abort)
- 100+ ready policy templates as YAML playbooks
- Multi-platform policies across Windows, Linux, macOS and Android
- Software management (install/uninstall, repos, LibreOffice, Python, PowerShell, client upgrade)
- Local user and group management (create/modify/delete, sudoers, SSH keys, domain join/leave, block/unblock)
- File and registry operations, service and scheduled-task management, power plans
- Security policies (certificate install, firewall rules, USB blocking)
- Self-service app store and policy portal
- Per-distro package-format handling (deb/rpm/apk)

**OS provisioning and migration**
- OS image management (Astra Linux, Alt Linux, RedOS, Windows)
- PXE / Kickstart deployment (TFTP/DHCP/ProxyDHCP, UEFI and BIOS, SMB)
- Windows to Linux migration with data and profile preservation
- Browser-data migration (Chrome/Yandex/Firefox/Edge)
- Batch migration jobs with per-host stage tracking

**Inventory and asset management**
- Asset catalog (software, OS, hotfixes, hardware, users, groups, paths)
- Hardware inventory (CPU/GPU/RAM/disk/NIC/monitor/encryption/security)
- Temporal asset history (versions, capacity)
- Configuration-change notifications

**Directory and identity beyond basic LDAP**
- FreeIPA and OpenLDAP in addition to Active Directory
- Multi-domain, TLS/cert, custom LDAP filters, user/group/host sync
- KeyCloak

**Multi-tenancy and infrastructure**
- Organizations and tenants
- Row-level security
- Clustering with a master server and replication via Redis pub/sub
- Postgres-based task queue

**Access control and security**
- Dynamic groups (mask / SQL / filter by device facts)
- Blacklist and whitelist by IP, network, UUID, user and host
- Geolocation-based access (CIDR to geo constraints)
- Key management (cloakkey / keygen)
- mTLS with CSR enrollment

**Compliance, audit and reporting**
- Syslog CEF export with 60+ event types and regex log parsers
- Custom report templates (PDF / XLSX)
- Scheduled report generation and delivery
- Apache Superset analytics (dashboards, charts, maps)
- RFC-5545 (iCalendar) RRULE scheduling
- Multi-channel alerting (Email / SMS / Telegram / MAX)

**Integrations**
- Ticketing with Jatoba and Kaspersky KSC

---

On the "made millions" point: whatever the revenue is, this isn't a thin rebrand collecting easy money. Running it is a real operation. We have 50+ engineers, and a geo-distributed bridge network that carries more than a petabyte of traffic a day at over 1.5M concurrent connections. The scale infrastructure and the management platform on top of it are ours, and they cost what infrastructure at that scale costs. That doesn't settle the licensing question, which is exactly why we just published the derived client. It only speaks to the idea that this was free to build.

Reasonable people can disagree about licensing on the derived client, and that's fair, which is why we've now published it. But "they just rebadged RustDesk" doesn't hold up against the list above. Happy to keep going on specifics with anyone arguing in good faith.

**Reply by ethaniel** [reply]
Wow, a full-on admission of guilt. Totally loving it. Keep talking.

**Reply by ermak-dev** [reply]
And a constructive discussion ends in low-effort trolling. Pathetic.

**Comment by clustev** [comment]
There is a clear timeline discrepancy: RustDesk Server was already publicly available in 2020–2021, while RuDesktop’s first public release dates to April 2022. This contradicts the claim that RuDesktop’s server predates the public release of RustDesk Server source code.

**Reply by ermak-dev** [reply]
@clustev What was public in 2020-2021 was an early prototype of the server. It didn't support encrypting traffic, so it wasn't usable for our purposes. The server in its current form was only published on 2022-05 (the LICENSE commit, b0b1981)

So we wrote our own: WebSocket + TLS instead of RustDesk's custom crypto, our own NAT traversal, and mTLS for P2P. The protocols are incompatible, so a RustDesk client can't talk to our server, which wouldn't be the case with a fork.
